lib/base.ts

```diff
@@ -96,6 +96,11 @@ export function ident(n: ts.Node): string {
   return null;
 }
 
+export function isValidDartIdentifier(name: ts.StringLiteral) {
```

Comment:
this is probably somewhere else in the code. Search for "JS$ which I believe was the existing hack.

Reply:
facade_converter.ts has
```
export function fixupIdentifierName(text: string): string {
  return (FacadeConverter.DART_RESERVED_WORDS.indexOf(text) !== -1 ||
          FacadeConverter.DART_OTHER_KEYWORDS.indexOf(text) !== -1 || text.match(/^(\d|_)/)) ?
      DART_RESERVED_NAME_PREFIX + text :
      text;
}
```

I think it's probably best to make isValidDartIdentifier accept the actual text string and then make fixupIdentifierName call it as well.